packages/xrpl/src/Wallet/index.ts

```diff
@@ -466,6 +466,22 @@ class Wallet {
       txCopy.URI = txCopy.URI.toUpperCase()
     }
 
+    if (
+      txCopy.TransactionType === 'NFTokenMint' &&
+      typeof txCopy.URI === 'string' &&
+      txCopy.URI === ''
+    ) {
+      throw new ValidationError('NFTokenMint: URI must not be empty string')
+    }
+
+    if (
+      txCopy.TransactionType === 'NFTokenMint' &&
+      txCopy.hasOwnProperty('URI') &&
+      txCopy.URI === undefined
+    ) {
+      delete txCopy.URI
+    }
```

Comment:
See my comment in #2212 about making this more general.

Reply:
Link to it for easier reference: https://github.com/XRPLF/xrpl.js/issues/2212#issuecomment-1434942602